### DISCUSSION TITLE
How to rename folder

### DISCUSSION THREAD

**Discussion by No1e** [pnp/pnpcore#1081] [Q&A] [answered]
Hi folks,

is there a way to rename folder in SharePoint by using PnP Core library? I am using latest version of PnP Core library.

Thanks,
Nole

**Comment by patrikhellgren** [comment, marked as answer]
You need to "move" the folder to rename it. Here is an example of how to do it:

``` C#
string folderUrl = $"{context.Uri.PathAndQuery}/Shared Documents/Test";

// Get a reference to the folder
IFolder testFolder = await context.Web.GetFolderByServerRelativeUrlAsync(folderUrl);

// Move the folder to rename it
await testFolder.MoveToAsync($"{context.Uri.PathAndQuery}/Shared Documents/Test2");
```

**Reply by jansenbe** [reply]
@patrikhellgren : thanks! I've also just added `Rename` methods to make renaming easier and more efficient.

**Comment by No1e** [comment]
Hi Patrik,

thank you very much.

Best regards,
Nole
